### PR TITLE
tonal auto masking

### PR DESCRIPTION
**Pull request by tzipisheinin**

### CODE REVIEW

**Comment by tzipisheinin** [comment]
[Tonal Test Masking](https://trello.com/c/vQwCZfNV/2441-tonal-test-masking)

File: src/cljs/audyx/audio/ambiences/masking.cljs [hunk 1]
```diff
@@ -128,9 +128,9 @@
           source-node (:head source-nodes)
           monitor-source-node (:head monitor-nodes)
           volume-node (audio-context/create-gain-node)
-          transducer (cabin/masking-transducer)
+          transducer (or transducer (cabin/masking-transducer))
```
[the remark this answers is not in the document]
Comment (tzipisheinin) [reply src/cljs/audyx/audio/ambiences/masking.cljs]: I saved it for audyx code (our original plan audyx and technician will use the deps)
but maybe audyx will send this data too

File: src/cljs/audyx/audio_engine/tonal_auto_masking.cljs [hunk 2]
```diff
@@ -43,16 +43,20 @@
   [tonal-tests]
   (let [comparator (fn [a b] (if (or (nil? a) (nil? b)) (compare a b) (compare b a)))]
     (some->>
+      tonal-tests
```
[the remark this answers is not in the document]
Comment (tzipisheinin) [reply src/cljs/audyx/audio_engine/tonal_auto_masking.cljs]: sending tonal tests it is only in deps
so I guess it was forgotten

File: src/cljs/audyx/audio_engine/tonal_auto_masking.cljs [hunk 3]
```diff
@@ -43,16 +43,20 @@
...
       (filter #(not-free-field? %))
       (filter #(threshold? %))
       (sort-by :done_at comparator))))
 
 ; Get latest none FF tests
+; check if
 (defn get-latest-tests [{{:keys [ears]} :hearing-context} tonal-tests]
-  (when-let [filtered-tests (no-ff-tests tonal-tests)]  
+  (if-let [filtered-tests (no-ff-tests tonal-tests)]
```
[the remark this answers is not in the document]
Comment (tzipisheinin) [reply src/cljs/audyx/audio_engine/tonal_auto_masking.cljs]: just first iteration of the above solution
I will delete it

File: src/cljs/audyx/test/auto_masking.cljs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+(defn get-full-yaml [yaml transducer]
+  (let [yaml-clj (js->clj yaml :keywordize-keys true)
+        transducer-clj (js->clj transducer :keywordize-keys true)
+        transducer-definition (:transducer_definition transducer-clj)]
+    (-> (assoc yaml-clj :transducer_definition transducer-definition)
+        (assoc-in [:masking :transducer_definition] transducer-definition))))
+
+(defn ^:export playMask [yaml transducer frequency volume]
+  (let [full-yaml (get-full-yaml yaml transducer)]
```
[the remark this answers is not in the document]
Comment (tzipisheinin) [reply src/cljs/audyx/test/auto_masking.cljs]: same as if yaml is empty
what do you mean what happens?

File: src/cljs/audyx/audio/ambiences/masking.cljs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ItamarShDev) [reply src/cljs/audyx/audio/ambiences/masking.cljs]: exactly
the cabin is a component from audyx
not related to deps anymore

File: src/cljs/audyx/test/auto_masking.cljs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ItamarShDev) [reply src/cljs/audyx/test/auto_masking.cljs]: whouldnt it be `when-let`?

**Pull request closed without merging**